src/course7/homework/Challenge12.java

```diff
@@ -26,18 +26,21 @@ public static void main(String[] args) {
 
             sum += number;
 
-            while (true) {
-                System.out.println("Do you want to stop? Type Yes or No.");
+            while (true) { //I used the "true" condition to purposely send the "while" loop to infinite runs.
+                System.out.println("Do you want to stop? Type Yes or No."); //this asks the users if he/she wants the adding to stop.
 
                 String answer = scan.nextLine();
 
-                if (answer.equalsIgnoreCase("Yes")) {
+                if (answer.equalsIgnoreCase("Yes")) { //if the answer is "yes", regardless of case, then the program will exit this loop
+                    //and go back to the main "while" loop and ask for another number".
                     System.out.println("Your total is: " + sum);
-                    return;
-                } else if (answer.equalsIgnoreCase("No")) {
-                    break;
+                    return; //this command tells the code to go to the main "while" loop.
```

Comment:
as previously mentioned, this will not go to the parent while, but exits from the entire `main` method.